trio_run_in_process/run_in_process.py

```diff
@@ -223,3 +224,79 @@ async def open_worker_process() -> AsyncIterator[WorkerProcessAPI]:
                 yield worker
     finally:
         worker._dead = True
+
+
+class WorkerPool(WorkerPoolAPI):
+    def __init__(self, num_workers: int) -> None:
+        self._num_workers = num_workers
+        self._workers: List[WorkerProcessAPI] = []
+        self._busy_workers: List[WorkerProcessAPI] = []
+        self._exit_stack = AsyncExitStack()
```

Comment:
I'm thinking that this object should be managed by the `open_worker_pool` function so that we can have better guarantees about lifecycle management.

```python
def open_worker_pool():
    async with ExitStack() as stack:
        pool = WorkerPool(stack, ...)
        yield pool
```

This gives us the guarantee that the stack gets exited when the context exits.

Reply:
I like the idea but I ended up implementing it as a ctxmanager method: `WorkerPool._open()`, so that the exit stack stays an implementation detail of the pool